bot/exts/fun/quack.py

```diff
@@ -0,0 +1,62 @@
+from typing import Optional
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+API_URL = 'https://quackstack.pythondiscord.com'
+
+
+class Quack(commands.Cog):
+    """Cog for the .quack command."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    @commands.command()
+    async def quack(self, ctx: commands.Context, ducktype: Optional[str] = "duck", seed: Optional[str] = None) -> None:
+        """
+        Uses the Quackstack API to generate a random duck.
+
+        If a seed is provided, a duck is generated based on the given seed.
+        If a duck type is provided, either a duck or manduck will be generated.
+        """
+        ducktype = ducktype.lower()
+        if ducktype not in ["duck", "manduck"]:
+            raise commands.BadArgument(
+                "The duck type can only be `duck` or `manduck`."
+            )
```

Comment:
This could be a converter that returns an enum.

Reply:
Literal should work as a converter with the new d.py

Reply:
I- what

Reply:
This can be removed as it's now validated by the literal converter

Reply:
@Kronifer like @Numerlor said here, this can be removed as we have moved validation into discord.py's hands.